app/models/manageiq/providers/microsoft/infra_manager.rb

```diff
@@ -133,4 +155,13 @@ def build_connect_params(options)
 
     connect_params
   end
+
+  def host_handle(vm, options)
```

Comment:
So, if we want to get the handle associated with the VM's host - `vm.host` - then this method should be defined in the `vm.host` object. Then line 108 above becomes: `host_handle = vm.host.host_handle(options)`.

In addition, doesn't the authentication information come from what's defined in `vm.host`? If so, then there should be no need to pass in `options`, resulting in: `host_handle = vm.host.host_handle`.